app/src/main/java/nu/yona/app/api/manager/network/BaseImpl.java

```diff
@@ -253,9 +253,9 @@ void onError(retrofit2.Response response, DataLoadListener listener)
 				Converter<ResponseBody, ErrorMessage> errorConverter =
 						getRetrofit().responseBodyConverter(ErrorMessage.class, new Annotation[0]);
 				ErrorMessage errorMessage = errorConverter.convert(response.errorBody());
-				if (errorMessage != null && errorMessage.getCode() != null)
+				if (errorMessage != null)
 				{
-					if (ServerErrorCode.USER_NOT_FOUND.equals(errorMessage.getCode()))
+					if (errorMessage.getCode() != null && ServerErrorCode.USER_NOT_FOUND.equals(errorMessage.getCode()))
```

Comment:
Put these two in a method like this:
```
private static boolean isUserNotFoundError(ErrorMessage errorMessage )
{
    return errorMessage.getCode() != null && ServerErrorCode.USER_NOT_FOUND.equals(errorMessage.getCode());
}
```
And call it here as well as at line 397.

Reply:
shall I put that in a new ErrorUtil class ?

Reply:
Looking at it again, I see the method is not necessary, nor is the extra condition.
This is sufficient:
```
if (ServerErrorCode.USER_NOT_FOUND.equals(errorMessage.getCode()))
```
If ``errorMessage.getCode()`` returns ``null``, then ``equals`` will return ``false``.

Reply:
just done the same